Make the todo example works out of the box

poupougnac opened the pull request:
The todo example needs some files built with webpack. I've slightly change the main package.json to permit a global `npm run build`.

poupougnac commented:
I have also added the possibility to add a custom HTTP server for the websocket (for people using Express).

piglovesyou commented:
Thank you. Indeed it needs the way to build all sub project at once.

- The root directory should not have dependency on `webpack` that `client` already depends on.
- `runtests.sh` does the similar thing to what you wrote on `build` script. Needs refactor.
- I guess it's the best if `postinstall` does build them all so that a user doesn't need to hit a custom script, `build` .

Better idea? If no, still OK, I'll think about it after merging.

Besides that, adding `server` option to [the document](https://github.com/paldepind/synceddb/blob/master/API.md#new-serveroptions) would be appreciated.

poupougnac commented:
I fixed the unnecessary dependency to webpack in the main package.json. I agree with the prepublish hook, it's a good idea. Seems okay this way ?
And of course I've also updated the doc.

piglovesyou commented on examples/todo/README.md:
````diff
@@ -6,12 +6,15 @@ Try it
 Get dependencies
 ```
-npm install
+npm run build
````
Then here should be ...

poupougnac commented:
Sorry I forgot to update the exemple README. Should be okay.

piglovesyou commented:
It is great. Thanks again. I think I'll create another example using express based on your option.

poupougnac commented:
Great idea ! Thank you for your reactivity 👍
